src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -68,8 +64,8 @@ class OpenShiftDSL implements Serializable {
             "rc"  : "replicationcontroller",
             "dc"  : "deploymentconfig" ]
 
-    enum ContextId implements Serializable{
-        WITH_CLUSTER("openshift.withCluster"), WITH_PROJECT("openshift.withProject"), DO_AS("openshift.doAs")
+    enum ContextId implements Serializable {
+        WITH_CLUSTER("openshift.withCluster"), WITH_PROJECT("openshift.withProject"), WITH_CREDENTIALS("openshift.withCredentials/openshift.doAs")
```

Comment:
curious in how this works vs. the need for both `DO_AS("openshift.doAs")` and `WITH_CREDENTIALS("openshift.withCredentials")`

can you elaborate @coreydaley ?

Reply:
Even though doAs still exists, it is actually creating the new context as withCredentials (on line 327), so DO_AS is never referenced anywhere.  doAs was only left because there will be existing scripts that still use it.

Reply:
OK, and the `openshift.withCredentials/openshift.doAs` contents, the `ContextId` ... is this just an arbitrary key then, where its format has no bearing elsewhere, right?